device.c

```diff
@@ -942,8 +939,8 @@ void destroy_vcam_device(struct vcam_device *vcam)
 
     if (vcam->sub_thr_id)
         kthread_stop(vcam->sub_thr_id);
-    vcam_in_queue_destroy(&vcam->in_queue);
-    destroy_framebuffer(vcam->vcam_fb_fname);
+    vcamfb_destroy(vcam);
+    vfree(vcam->fb_priv);
```

Comment:
Can you combine the two operations into single function? That is, we don't have to invoke the sequence:
```c
vcamfb_destroy(vcam);
vfree(vcam->fb_priv);
```

Instead, single `vcamfb_destroy(vcam)` can release the allocated resources without explicitly passing `vcam->fb_priv`.